pyomo/core/base/initializer.py

```diff
@@ -316,6 +349,18 @@ def __call__(self, parent, idx):
             return self._fcn(parent, idx)
 
 
+class ParameterizedIndexedCallInitializer(IndexedCallInitializer):
+    """IndexedCallCallInitializer that accepts additional arguments"""
+
+    __slots__ = ()
+
+    def __call__(self, parent, idx, *args):
+        if idx.__class__ is tuple:
+            return self._fcn(parent, *args, *idx)
+        else:
+            return self._fcn(parent, *args, idx)
```

Comment:
I think I'm answering my own question by noticing that the argument ordering is indeed swapped here.

Reply:
@jsiirola there isn't really anything for you to resolve in my comments/questions, I just wanted to document my initial confusion. I am curious why the argument ordering and subsequent reordering was necessary. I'm guessing it had to do with having more control over the format of the index being passed in.

Reply:
Yeah - this turned into a bit of a mess (and yes, it is confusing that they were reversed).  The calls you are seeing here are all internal to the Initializer methods, so the user is never aware of them.  In this case, because the "Parameterized" initializer was new, the old `fcn` API was just `(block, index)`.  Parameterized was adding "value", so it made sense for the extra arg to come at the end.  This was further confused by the historical behavior where we would expand the value arg and pass it to the user-provided callback (leading things like the messy deprecation paths around `set.py` lines 1482-1520), and leaving things like this seemed to require less code gymnastics to implement the deprecation paths and make things compatible with the rest of the Initializer system.